cmd/node/factory/structs.go

```diff
@@ -1753,21 +1748,26 @@ func newMetaBlockProcessorAndTracker(
 		return nil, nil, err
 	}
 
-	metaProcessor, err := block.NewMetaProcessor(
-		coreServiceContainer,
-		state.AccountsAdapter,
-		data.MetaDatapool,
-		forkDetector,
-		shardCoordinator,
-		nodesCoordinator,
-		specialAddressHandler,
-		core.Hasher,
-		core.Marshalizer,
-		data.Store,
-		shardsGenesisBlocks,
-		requestHandler,
-		core.Uint64ByteSliceConverter,
-	)
+	argumentsBaseProcessor := block.ArgBaseProcessor{
+		Accounts:              state.AccountsAdapter,
+		ForkDetector:          forkDetector,
+		Hasher:                core.Hasher,
+		Marshalizer:           core.Marshalizer,
+		Store:                 data.Store,
+		ShardCoordinator:      shardCoordinator,
+		NodesCoordinator:      nodesCoordinator,
+		SpecialAddressHandler: specialAddressHandler,
+		Uint64Converter:       core.Uint64ByteSliceConverter,
+		StartHeaders:          shardsGenesisBlocks,
+		RequestHandler:        requestHandler,
+		Core:                  coreServiceContainer,
+	}
+	arguments := block.ArgMetaProcessor{
```

Comment:
I don't thing there should be a discrepancy between how we pass the `ArgBaseProcessor` and `ArgMetaProcessor`. What I mean by this, is that `ArgMetaProcessor` is passed to the constructor by value, and `ArgBaseProcessor` is a pointer. I would use the same approach for both, either pointer, either value. Since they are structures containing only interface members and pointers, I would pass them both by value

Reply:
Done. Now that structures are passed by value